src/Anonymization/Anonymizer/AbstractEnumAnonymizer.php

```diff
@@ -67,7 +67,7 @@ public function anonymize(Update $update): void
                 ->raw(
                     '? + 1 = ?',
                     [
-                        $expr->mod($this->getJoinColumn(), $sampleCount),
+                        $expr->mod($this->getJoinColumn(), \min($targetCount, $sampleCount)),
```

Comment:
Let's assume we have a target table with 10 rows and a sample table with 20 rows.

When joining the sample table, we only join a number of rows equals to the target table count, and we number each of these rows from 1. On both sides of the join, we therefore have the same count of rows numbered the same way, i.e. from 1 to 10.

Because the modulo operation present in the join condition uses the sample table count as divisor, which here equals 20, the condition will encounter the following case : `MOD(10, 20) + 1 = "sample_table"."rownum"`. The operation `MOD(10, 20) + 1` will be equal to 11, but there is no 11 numbered row in the right side of the join. As we perform a `left join`, the join will in that particular case result in a `null` sample value.

An error occurs when `null` is not allowed for the target column.

That's why the modulo operation must always use the count of the smallest table as divisor to restore the expected behavior : `MOD(10, 20) + 1` becomes `MOD(10, 10) + 1` which equals 1.

Reply:
Hum, you are probably right, but I was sure I tested this exact use case before, hence the `mod()` operator... I must have forgotten to implement the counterpart in this single place.

You should look at the "multi column anonymizer" as well to be sure the join is correct there too.

Reply:
> You should look at the "multi column anonymizer" as well to be sure the join is correct there too.

Good point! I fix it as well.

Reply:
Fix added to the multi column anonymizer :+1: